frontend/packages/ceph-storage-plugin/src/components/create-storage-system/reducer.ts

```diff
@@ -0,0 +1,86 @@
+import { BackingStorageType, RHCS } from '../../constants/create-storage-system';
+
+export type WizardState = React.ReducerState<WizardReducer>;
+export type WizardDispatch = React.Dispatch<React.ReducerAction<WizardReducer>>;
+
+export type WizardCommonProps = {
+  state: WizardState;
+  dispatch: WizardDispatch;
+};
+
+/* State of CreateStorageSystem */
+export const initialState: CreateStorageSystemState = {
+  currentStep: {
+    count: 1,
+  },
+  backingStorage: {
+    type: BackingStorageType.EXISTING,
+    externalProvider: RHCS,
+  },
+  storageClass: { name: '', provisioner: '' },
+};
+
+type CreateStorageSystemState = {
+  currentStep: StepState;
+  backingStorage: {
+    type: BackingStorageType;
+    externalProvider: string;
+  };
+  storageClass: { name: string; provisioner?: string };
+};
+
+type StepState = {
+  count: number;
+};
+
+/* Reducer of CreateStorageSystem */
+export const reducer: WizardReducer = (prevState, action) => {
+  const newState = { ...prevState };
+  switch (action.type) {
+    case 'currentStep/incrementCount':
+      newState.currentStep = { count: newState.currentStep.count + 1 };
+      break;
+    case 'currentStep/resetCount':
+      newState.currentStep = { count: 1 };
+      break;
+    case 'backingStorage/setType':
+      newState.backingStorage = {
+        ...prevState.backingStorage,
+        type: action.payload,
+      };
+      break;
+    case 'backingStorage/setExternalProvider':
+      newState.backingStorage = {
+        ...prevState.backingStorage,
+        externalProvider: action.payload,
+      };
+      break;
+    case 'wizard/setStorageClass':
+      newState.storageClass = {
+        name: action.payload.name,
+        provisioner: action.payload?.provisioner,
+      };
+      break;
+    default:
+      throw new TypeError(`${action} is not a valid reducer action`);
+  }
+  return newState;
+};
+
+export type WizardReducer = (
+  prevState: CreateStorageSystemState,
+  action: CreateStorageSystemAction,
+) => CreateStorageSystemState;
+
+type CreateStorageSystemAction =
+  | { type: 'currentStep/incrementCount' }
+  | { type: 'currentStep/resetCount' }
+  | { type: 'backingStorage/setType'; payload: CreateStorageSystemState['backingStorage']['type'] }
+  | {
+      type: 'backingStorage/setExternalProvider';
+      payload: CreateStorageSystemState['backingStorage']['externalProvider'];
+    }
+  | {
+      type: 'wizard/setStorageClass';
```

Comment:
There are various instances where we set the storage class.
So it will help to understand that this is global, may be I can say `global/storageClass`

Reply:
No I don't think its sounds good,